labs/a1.md

```diff
@@ -37,50 +37,87 @@ or anything of the sort to worry about (which will also be true for the followin
-by making directories for each of them, specifically for my favorite artist Future.
+`#!` is a human-readable representation of a [magic number][1] `0x23 0x21` 
+which can tell the shell to pass execution of the rest of the file to a 
+specified interpreter. If your script is run as an executable (e.g. 
+`./awesome_shell_script`) with a shebang line, then the shell will invoke the 
+executable (usually an interpreter) at `path/to/interpreter` to run your 
+script. If your script is passed as an argument to an interpreter e.g. `bash 
+awesome_shell_script`, then the shebang has no effect and `bash` will handle 
+the script's execution.
 
-I have hosted a list of my favorite albums followed by their respective artist at 
-`https://raw.githubusercontent.com/0xcf/decal-labs/master/a1/albums.txt`,
-in a comma delimited format like `Die Lit, Playboi Carti`. For the GOAT artist Future,
-I want to create a folder for each of his albums. For example, for an entry like
-`SUPER SLIMEY, Future` I would expect a directory called `SUPER SLIMEY` to be created.
+**Why is this important?** The shebang line can be considered a useful piece of 
+metadata which passes the concern of _how_ a script is executed from the user 
+to the program's author. `awesome_shell_script` could be a `bash` script, a 
+`python` script, a `ruby` script, etc. The idea is that only the script's 
+behavior, not its implementation details, should matter to the user who calls 
+the script.
 
-**TLDR**: You need to fetch the list from the web, filter out the albums we want,
-trim out the album name, and then make a directory for each one, all in one line.
+You may have seen some variant of `#!/bin/sh`. Although initially referencing 
+the Bourne shell, on modern systems `sh` has come to reference to the [Shell 
+Command Language][2], which is a POSIX specification with many implementations. 
+`sh` is usually symlinked to one of these POSIX-compliant shells which 
+implement the Shell Command Language. On Debian, for instance, `sh` is 
+symlinked to the shell `dash`. It is important to note that `bash` does **not** 
+comply with this standard, although running `bash` as `bash --posix` makes it 
+more compliant.
+
+### Piping
+
+We can use the `|` character to chain multiple commands together in one line. For instance: `command1 | command2`
+will pass the output of `command1` as the input to `command2`. We can repeat this as many times as we need.
+
+### Looping with `for`
+
+Bash can repeat an operation on multiple objects with a for loop. The syntax is as follows:
 
 ```bash
-$ cat albums.txt
-...
-Drip or Drown 2, Gunna
-Playboi Carti, Playboi Carti
-DS2 (Deluxe), Future  <-- GOAT album detected!
-Drip Harder, Lil Baby
-The WIZRD, Future  <-- GOAT album detected!
-What a Time To Be Alive, Drake
-...
+for VARIABLE in LIST; do
 
-# After our magic one liner...
+    CONSEQUENT-COMMANDS
 
-$ ls
-'DS2 (Deluxe)'   'The WIZRD'   ...
-# We got our new directories!
+done
```

Comment:
maybe make a note to look at the a1 participation assignment for an example of `for` usage